### ISSUE TITLE
Support Consul 1.7 consecutive pass/fail on health checks

### ISSUE THREAD

**Issue by tgross** [hashicorp/nomad#6913]
Consul 1.7 provides support for configuring health checks [to become passing/critical only if a specified number of consecutive checks return passing/critical](https://www.consul.io/docs/agent/checks.html#success-failures-before-passing-critical
). Nomad service health check config should also allow this configuration.

**Comment by shantanugadgil** [comment]
What ... postponed again! 😭

**Comment by shoenig** [comment]
Sorry about the postponement @shantanugadgil , this is blocked by
https://github.com/hashicorp/consul/issues/7764
(Consul does not currently accept configuration of these fields via HTTP API)

**Comment by shoenig** [comment]
Looks like https://github.com/hashicorp/consul/pull/7949 got merged for Consul v1.8.1

**Comment by shantanugadgil** [comment]
🎉
